### PR TITLE
[doc] add user documentation for UPDATE.

### PR DESCRIPTION
**Pull request by liming30**
### Purpose

add user documentation for `UPDATE`.  #787 

### Tests

No.

### API and Format

No.

### Documentation

No.

### CODE REVIEW

File: docs/content/how-to/writing-tables.md [hunk 1]
```diff
@@ -234,6 +234,47 @@ For more information of drop-partition, see
 
 {{< /tabs >}}
 
+## Updating tables
+Currently, Paimon supports updating records by using `UPDATE` in Flink. You can perform `UPDATE` in Flink's `batch` mode.
```
Comment (FangYongs) [inline docs/content/how-to/writing-tables.md:238]: Use  "in Flink 1.17 and later versions."?

Comment (liming30) [reply docs/content/how-to/writing-tables.md]: resolved.

**Comment by JingsongLi** [comment]
Hi @liming30 , if this PR is the last PR for the issue. Please use https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
Just like `closes https://github.com/apache/incubator-paimon/issues/787`.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**